test/e2e/specs/editor/various/block-switcher.spec.js

```diff
@@ -0,0 +1,191 @@
+		await page.evaluate( () => {
+			[
+				'core/quote',
+				'core/pullquote',
+				'core/paragraph',
+				'core/group',
+				'core/heading',
+				'core/columns',
+			].forEach( ( block ) => wp.blocks.unregisterBlockType( block ) );
+		} );
+
+		// Verify the block switcher exists.
+		expect(
+			page.locator(
+				'.block-editor-block-toolbar .block-editor-block-switcher'
+			)
+		).toBeTruthy();
+
+		// Verify the correct block transforms appear.
+		expect( await getAvailableBlockTransforms() ).toHaveLength( 0 );
+	} );
+
+	test.describe( 'Conditional tranformation options', () => {
+		test.describe( 'Columns tranforms', () => {
+			test( 'Should show Columns block only if selected blocks are between limits (1-6)', async ( {
+				page,
+				editor,
+				blockSwitcher,
+			} ) => {
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( '- List content' );
+				await page.keyboard.press( 'ArrowUp' );
+				await editor.insertBlock( {
+					name: 'core/heading',
+				} );
+				await page.keyboard.type( 'I am a header' );
+				await page.keyboard.down( 'Shift' );
+				await page.keyboard.press( 'ArrowUp' );
+				await page.keyboard.up( 'Shift' );
+				expect(
+					await blockSwitcher.getAvailableBlockTransforms()
+				).toEqual( expect.arrayContaining( [ 'Columns' ] ) );
+			} );
+			test( 'Should NOT show Columns transform only if selected blocks are more than max limit(6)', async ( {
+				page,
+				editor,
+				pageUtils,
+				blockSwitcher,
+			} ) => {
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( '- List content' );
+				await page.keyboard.press( 'ArrowUp' );
+				await editor.insertBlock( {
+					name: 'core/heading',
+				} );
+				await page.keyboard.type( 'I am a header' );
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( 'First paragraph' );
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( 'Second paragraph' );
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( 'Third paragraph' );
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( 'Fourth paragraph' );
+				await page.keyboard.press( 'Enter' );
+				await page.keyboard.type( 'Fifth paragraph' );
+				await pageUtils.pressKeys( 'primary+a' );
+				await pageUtils.pressKeys( 'primary+a' );
+				expect(
+					await blockSwitcher.getAvailableBlockTransforms()
+				).not.toEqual( expect.arrayContaining( [ 'Columns' ] ) );
+			} );
+		} );
+	} );
+} );
+
+class blockSwitcher {
+	constructor( { page } ) {
+		this.page = page;
+	}
+
+	async getAvailableBlockTransforms() {
+		await this.page.click( 'role=button[name="List"i]' );
+		return this.page.evaluate( ( buttonSelector ) => {
+			return Array.from(
+				document.querySelectorAll( buttonSelector )
+			).map( ( button ) => {
+				return button.textContent;
+			} );
+		}, '.block-editor-block-switcher__popover .block-editor-block-switcher__transforms__menugroup button' );
+	}
```

Comment:
We can rewrite this into accessible selectors:

```js
await this.getBlockSwitcher( blockName ).click();
return await this.page.getByRole( 'menu', { name: blockName } )
	.getByRole( 'menuitem' )
	.allTextContents();
```

I left out the implementation of `getBlockSwitcher` here though. Feel free to ask questions if it's not clear!